src/main/java/org/tron/core/db/TronStoreWithRevoking.java

```diff
@@ -1,71 +1,114 @@
 package org.tron.core.db;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Iterator;
+import java.util.Map;
 import java.util.Objects;
+
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Streams;
+import com.google.common.reflect.TypeToken;
+import com.google.protobuf.InvalidProtocolBufferException;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.ArrayUtils;
 import org.tron.core.capsule.ProtoCapsule;
 import org.tron.core.db.AbstractRevokingStore.RevokingTuple;
+import org.tron.core.db2.common.IRevokingDB;
+import org.tron.core.db2.core.RevokingDBWithCachingNewValue;
+import org.tron.core.db2.core.RevokingDBWithCachingOldValue;
+import org.tron.core.exception.BadItemException;
+import org.tron.core.exception.ItemNotFoundException;
 
 @Slf4j
 public abstract class TronStoreWithRevoking<T extends ProtoCapsule> extends TronDatabase<T> {
 
-  private RevokingDatabase revokingDatabase;
+  protected IRevokingDB revokingDB;
+  private TypeToken<T> token = new TypeToken<T>(getClass()) {};
 
   protected TronStoreWithRevoking(String dbName) {
-    this(dbName, RevokingStore.getInstance());
+    if (true) {

```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Remove this if statement. ([squid:S1145](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS1145))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1159&issues=AWSrvNJUl6vpliCetG6S&open=AWSrvNJUl6vpliCetG6S)

[ISSUE_KEY]: AWSrvNJUl6vpliCetG6S